Source/monster.cpp

```diff
@@ -1356,7 +1355,7 @@ bool MonsterWalk(int i, MonsterMode variant)
 		case MonsterMode::MoveSideways:
 			dMonster[monster.position.tile.x][monster.position.tile.y] = 0;
 			monster.position.tile = { monster._mVar1, monster._mVar2 };
-			dFlags[monster.position.temp.x][monster.position.temp.y] &= ~BFLAG_MONSTLR;
+			// Don't delete this (in devilutionX) unncessary assignment. It's needed to stay compatible with vanilla saves.

```

Comment:
I feel the comment would be better if it described what the consequences is

```suggestion
			// dMonster is set here for backwards comparability, without it the monster would be invisible is save is loaded in older version
```